### PR TITLE
refactor: wallet events

### PR DESCRIPTION
**Pull request by alexfreska**
- Wallet events have been updated to use the new types and data format. https://github.com/SiaFoundation/walletd/issues/134

### CODE REVIEW

File: libs/types/src/v2.ts [hunk 1]
```diff
@@ -94,4 +96,48 @@ export type AttestationElement = StateElement & {
...
+      renterRollover: Currency
+      hostRollover: Currency
+      renterSignature: Signature
+      hostSignature: Signature
+    }
+  }
+
+export type V2FileContractResolutionDataStorageProof =
+  V2FileContractResolutionTypeBase & {
+    type: 'storage proof'
```
Comment (alexfreska) [inline libs/types/src/v2.ts:129]: @n8maninger should this be camel case? wallet event types are camel case.

File: libs/types/src/v2.ts [hunk 2]
```diff
@@ -94,4 +96,48 @@ export type AttestationElement = StateElement & {
...
 
-export type V2FileContractResolutionType = unknown // TODO: replace `unknown` with the actual type
+type V2FileContractResolutionTypeBase = {
+  parent: V2FileContractElement
+}
+
+export type V2FileContractResolutionTypeExpiration =
+  V2FileContractResolutionTypeBase & {
+    type: 'expiration'
+    resolution: Record<string, never> // is this always empty for expiration type?
```
Comment (alexfreska) [inline libs/types/src/v2.ts:106]: @n8maninger could you confirm?

File: libs/types/src/v2.ts (hunk 1 above)
Comment (n8maninger) [reply libs/types/src/v2.ts:129]: I'd like to be consistent, but these are in `core`. For now we'll leave the inconsistency and discuss in a follow up if that's alright.

File: libs/types/src/v2.ts (hunk 2 above)
Comment (n8maninger) [reply libs/types/src/v2.ts:106]: That's correct, the resolution on an "expiration" type is empty

**Review by n8maninger — APPROVED** [review]

**Review by ChrisSchinnerl — APPROVED** [review]

**Comment by alexfreska** [comment]
### Merge activity

* **Jun 20, 12:40 PM EDT**: @alexfreska started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/SiaFoundation/web/644).

**Pull request merged**